operator/cmd/mesh/install.go

```diff
@@ -136,12 +138,17 @@ func runApplyCmd(cmd *cobra.Command, rootArgs *rootArgs, iArgs *installArgs, log
 	if err != nil {
 		return err
 	}
+	setFlags := applyFlagAliases(iArgs.set, iArgs.manifestsPath, iArgs.revision)
 	// Ignore the err because we don't want to show
 	// "no running Istio pods in istio-system" for the first time
 	_ = DetectIstioVersionDiff(cmd, tag, kubeClient, iArgs)
 	// Warn users if they use `istioctl install` without any config args.
 	if !rootArgs.dryRun && !iArgs.skipConfirmation {
-		prompt := fmt.Sprintf("This will install the Istio %s profile into the cluster. Proceed? (y/N)", tag)
+		profile, enabledComponents, err := getProfileAndEnabledComponents(setFlags, iArgs.inFilenames, iArgs.force, iArgs.kubeConfigPath, iArgs.context, l)
+		if err != nil {
+			return fmt.Errorf("failed to get profile and enabled components: %v", err)
```

Comment:
Do we really want to prevent a user from installing Istio when these components are not found (which is just use for informative messages)?

Reply:
`getProfileAndEnabledComponents` will return the same error from the `InstallManifests`, so if we ignore the error from pre-check, it will also failed in the `InstallManifests` step. Besides, what message we should tell to end users if the error is return in pre-check.